ci/PKGBUILD

```diff
@@ -0,0 +1,115 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+_realname=arrow
+pkgbase=mingw-w64-${_realname}
+pkgname="${MINGW_PACKAGE_PREFIX}-${_realname}"
+pkgver=0.13.0.9000
+pkgrel=8000
+pkgdesc="Apache Arrow is a cross-language development platform for in-memory data (mingw-w64)"
+arch=("any")
+url="https://arrow.apache.org/"
+license=("Apache-2.0")
+depends=("${MINGW_PACKAGE_PREFIX}-boost"
+         "${MINGW_PACKAGE_PREFIX}-double-conversion"
+         "${MINGW_PACKAGE_PREFIX}-thrift"
+         "${MINGW_PACKAGE_PREFIX}-zlib")
+makedepends=("${MINGW_PACKAGE_PREFIX}-cmake"
+             "${MINGW_PACKAGE_PREFIX}-gcc")
+options=("staticlibs" "strip" "!buildflags")
+source_dir=apache-${_realname}-${pkgver}
+source=("${source_dir}"::"git+https://github.com/apache/arrow")
+sha256sums=("SKIP")
+
+cmake_build_type=release
+cpp_build_dir=build-${CARCH}-cpp
+
+pkgver() {
+  cd "$source_dir"
+  grep Version r/DESCRIPTION | cut -d " " -f 2
+}
+
+prepare() {
+  pushd ${source_dir}
+  #patch -p1 -N -i ${srcdir}/3923.patch
+  popd
+}
+
+build() {
+  ARROW_CPP_DIR="$(pwd)/${source_dir}/cpp"
+  [[ -d ${cpp_build_dir} ]] && rm -rf ${cpp_build_dir}
+  mkdir -p ${cpp_build_dir}
+  pushd ${cpp_build_dir}
+
+  # Workaround to fix static libparquet
+  export CXXFLAGS="-DARROW_STATIC"
+
+  export CC="/C/Rtools${MINGW_PREFIX/mingw/mingw_}/bin/gcc"
+  export CXX="/C/Rtools${MINGW_PREFIX/mingw/mingw_}/bin/g++"
+  export PATH="/C/Rtools${MINGW_PREFIX/mingw/mingw_}/bin:$PATH"
+  export CPPFLAGS="-I${MINGW_PREFIX}/include"
+  export LIBS="-L${MINGW_PREFIX}/libs"
+
+  MSYS2_ARG_CONV_EXCL="-DCMAKE_INSTALL_PREFIX=" \
+    ${MINGW_PREFIX}/bin/cmake.exe \
+    ${ARROW_CPP_DIR} \
+    -G "MSYS Makefiles" \
+    -DCMAKE_INSTALL_PREFIX=${MINGW_PREFIX} \
+    -DCMAKE_BUILD_TYPE=${cmake_build_type} \
+    -DARROW_BUILD_STATIC=ON \
+    -DARROW_BUILD_TESTS=OFF \
+    -DARROW_PARQUET=ON \
+    -DARROW_PLASMA=OFF \
+    -DARROW_HDFS=OFF \
+    -DARROW_PYTHON=OFF \
+    -DARROW_BOOST_USE_SHARED=OFF \
+    -DARROW_WITH_SNAPPY=OFF \
+    -DARROW_WITH_ZSTD=OFF \
+    -DARROW_WITH_LZ4=OFF \
+    -DARROW_JEMALLOC=OFF \
+    -DARROW_BUILD_SHARED=OFF\
+    -DARROW_BOOST_VENDORED=OFF \
```

Comment:
This was deprecated in #4623 , you want ARROW_BOOST_SOURCE=SYSTEM. Or just ignore this flag for now.

Reply:
Ha, very recently deprecated :)

Ok, I'll update.

Reply:
FWIW `ARROW_BOOST_VENDORED` is also still used in https://github.com/apache/arrow/blob/master/cpp/Dockerfile.ubuntu-trusty#L61 and in https://github.com/apache/arrow/blob/master/cpp/Dockerfile.ubuntu-xenial#L69. Shall I update those too while I'm at it?

Reply:
Addressed the one here and fixed the others in https://github.com/apache/arrow/pull/4656